### PR TITLE
Some code refactoring

### PR DESCRIPTION
**Pull request by r4f4**
- Move repeating blocks of code out to funtions
- Move blocks of code out to a function for better reading

Signed-off-by: Rafael Fonseca <r4f4rfs@gmail.com>

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -603,15 +632,240 @@ apply_excludes (Pool       *pool,
...
+      pool_best_solvables (pool, &q, 0);
+      for (int j = 0; j < q.count; j++)
+        queue_push (&pile, q.elements[j]);
+    }
+
+  return pile;
+}
+
+static void
+output_resolved_packages (Pool       *pool,
```
Comment (igor-raits) [inline main.c]: i wouldn't split this to function. no one can reuse it.

File: main.c [hunk 2]
```diff
@@ -603,15 +632,240 @@ apply_excludes (Pool       *pool,
...
+        }
+    }
+  while (!all_tested);
+
+  return solv_failed;
+}
+
+static Queue
+pile_from_solvables (Pool  *pool,
+                     GStrv solvables)
```
Comment (igor-raits) [inline main.c]: one more space before solvables

File: main.c [hunk 3]
```diff
@@ -603,15 +632,240 @@ apply_excludes (Pool       *pool,
...
+    map_set (&modular_pkgs, *pp);
+
+  return modular_pkgs;
+}
+
+static gboolean
+_install_transaction (Pool         *pool,
+                      Queue        *pile,
+                      Queue        *job,
+                      unsigned int indent)
```
Comment (igor-raits) [inline main.c]: indent

File: main.c [hunk 4]
```diff
@@ -78,6 +78,118 @@ parse_module_requires (Pool       *pool,
...
+    /* Prv: modular-package() */
+    Id modpkg = pool_str2id (pool, MODPKG_PROV, 1);
+    solvable_add_deparray (s, SOLVABLE_PROVIDES, modpkg, 0);
+  }
+}
+
+static void
+add_module_rpm_artifacts (Pool           *pool,
+                          ModulemdModule *module,
+                          Id             sdep)
```
Comment (igor-raits) [inline main.c]: indent

File: main.c [hunk 5]
```diff
@@ -78,6 +78,118 @@ parse_module_requires (Pool       *pool,
...
+    Id require = parse_module_requires (pool, req);
+    requires = dep_or_rel (pool, requires, require, REL_OR);
+  }
+  solvable_add_deparray (solvable, SOLVABLE_REQUIRES, requires, 0);
+}
+
+static void
+add_artifacts_dependencies (Pool  *pool,
+                            Queue *sel,
+                            Id    sdep)
```
Comment (igor-raits) [inline main.c]: indent

**Comment by igor-raits** [comment]
apart from few comments - Im okay with this

**Comment by r4f4** [comment]
Thank you for the review. I've addressed the issues you mentioned.

**Review by igor-raits — COMMENTED** [review]
I didn't test this, so @lubomir should have last word. Style-wise looks good

**Comment by lubomir** [comment]
I ran a couple tests with this patch and it gives the same results as original code, so :+1: from me.

**Pull request merged**